core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -745,16 +709,54 @@ private[zio] final class FiberContext[E, A](
     }
   }
 
-  private[this] final def exitSupervision: UIO[Any] = ZIO.effectTotal(supervised.pop())
+  private[this] final def daemonEnter(flag: DaemonStatus): UIO[Unit] = ZIO.effectTotal {
+    daemonStatus.push(flag.toBoolean)
+  }
+
+  private[this] final def daemonExit: UIO[Unit] = ZIO.effectTotal {
+    val _ = daemonStatus.popOrElse(false)
+  }
+
+  final def propagateAncestorInterruption(): Unit = {
+    var fiber = self: FiberContext[_, _]
+
+    while (fiber ne null) {
+      addInterruptor(fiber.state.get.interrupted)
+
+      fiber = fiber.parentFiber
```

Comment:
It will need to be documented later that propagation is not hierarchical – a nested fiber will be interrupted even when all of its superiors are uninterruptible/unresponsive, as long as any one of them is targeted by kill0.

Reply:
Only if the children are interruptible, which by default they would not be, because interruptibility status is inheritable.

It's only if someone specifically made some section (or fiber) interruptible that the children would potentially terminate first, but that's the same as in-fiber semantics: if you call downstream code and there's an interruptible section in there, it could interrupt yourself unless you're dealing with it.

It's still worth documenting, of course.